pkg/scheduler/framework/plugins/nodeports/node_ports.go

```diff
@@ -92,12 +88,19 @@ func getPreFilterState(cycleState *framework.CycleState) (preFilterState, error)
 
 // Filter invoked at the filter extension point.
 func (pl *NodePorts) Filter(ctx context.Context, cycleState *framework.CycleState, pod *v1.Pod, nodeInfo *nodeinfo.NodeInfo) *framework.Status {
-	state, err := getPreFilterState(cycleState)
+	wantPorts, err := getPreFilterState(cycleState)
 	if err != nil {
 		return framework.NewStatus(framework.Error, err.Error())
 	}
-	_, reasons, err := predicates.PodFitsHostPortsPredicate(pod, state, nodeInfo)
-	return migration.PredicateResultToFrameworkStatus(reasons, err)
+
+	existingPorts := nodeInfo.UsedPorts()
+	for _, cp := range wantPorts {
+		if existingPorts.CheckConflict(cp.HostIP, string(cp.Protocol), cp.HostPort) {
+			return framework.NewStatus(framework.Unschedulable, ErrReason)
+		}
+	}
```

Comment:
Ditto. It's duplicated with "podFitsHostPorts()" in predicates.go.

Reply:
answered.